src/EditorFeatures/Test/Completion/CompletionServiceTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Completion
+{
+    [UseExportProvider]
+    public class CompletionServiceTests
+    {
+        [Fact]
+        public async Task TestNuGetCompletionProvider()
+        {
+            var code = @"
+using System.Diagnostics;
+class Test {
+    void Method() {
+        Debug.Assert(true, ""$$"");
+    }
+}
+";
+
+            using var workspace = TestWorkspace.CreateCSharp(code, openDocuments: true);
+
+            var nugetCompletionProvider = new DebugAssertTestCompletionProvider();
+            var reference = new MockAnalyzerReference(nugetCompletionProvider);
+            var project = workspace.CurrentSolution.Projects.Single().AddAnalyzerReference(reference);
+            var completionService = project.LanguageServices.GetRequiredService<CompletionService>();
+
+            var document = project.Documents.Single();
+            var caretPosition = workspace.DocumentWithCursor.CursorPosition ?? throw new InvalidOperationException();
+            var completions = await completionService.GetCompletionsAsync(document, caretPosition);
+
+            Assert.NotNull(completions);
+            var item = Assert.Single(completions.Items.Where(item => item.ProviderName == typeof(DebugAssertTestCompletionProvider).FullName));
+            Assert.Equal("Assertion failed", item.DisplayText);
+        }
+
+        private class MockAnalyzerReference : AnalyzerReference, ICompletionProviderFactory
```

Comment:
Do we need to add a test using a AnalyzerFileReference created from test assembly (and export the test provider)?

Reply:
I'll file a follow-up issue

Reply:
➡️ https://github.com/dotnet/roslyn/issues/43443